amd/src/app/block/element.js

```diff
@@ -514,6 +579,19 @@ export default class BlockElement {
         this.#element.querySelector('.sharing_cart_items').prepend(element);
 
         this.setupItem(element);
+
+        // Re-attach event listeners to the checkboxes
+        const checkboxSelector = '.sharing_cart_item input[data-action="bulk_select"][type="checkbox"]';
+        const getItemCheckboxes = () => this.#element.querySelectorAll(checkboxSelector);
+
+        getItemCheckboxes().forEach(checkbox => checkbox.addEventListener('change', () => {
+            this.updateBulkDeleteButtonState();
+            this.updateSelectAllState();
+        }));
+
+        // Update the state of the select all checkbox and bulk delete button
+        this.updateBulkDeleteButtonState();
+        this.updateSelectAllState();
```

Comment:
This part won't work. - Remember to test your code ;)

The reason why is that currently there's no such thing as `this.updateBulkDeleteButtonState();` & `this.updateSelectAllState();`. The ones i think you're refering to are not accessible like that. Only class functions & properties are accessible using `this.something`

You should probably also move this logic to another function perhaps the `setupItem` function? The "renderItem" function should only render the item, while setupItem or something else should do all the stuff afterwards.

You should move some of the functions out into the class, this would make you able to reuse them.

Reply:
It seems to work fine with and without the code. So i guess it should just be deleted.

Reply:
No, new backups will not work with the select all button without refreshing the page. They won't have the eventlisteners added since you only add them  to the initial ones, so this code is still required. It should just be fixed